src/checkers/inference/solver/backend/encoder/AbstractConstraintEncoderFactory.java

```diff
@@ -1,16 +1,20 @@
 package checkers.inference.solver.backend.encoder;
 
+import checkers.inference.solver.backend.FormatTranslator;
 import checkers.inference.solver.frontend.Lattice;
 import checkers.inference.util.ConstraintVerifier;
+import checkers.inference.solver.backend.z3.encoder.Z3BitVectorSubtypeConstraintEncoder;
+import checkers.inference.solver.backend.maxsat.encoder.MaxSATImplicationConstraintEncoder;
 
 /**
  * Abstract base class for all concrete {@link ConstraintEncoderFactory}. Subclasses of {@code AbstractConstraintEncoderFactory}
  * should override corresponding createXXXEncoder methods to return concrete implementations if the
- * {@link checkers.inference.solver.backend.SolverType} that the subclasses belong to support encoding such constraints.
+ * solver backend that the subclasses belong to support encoding such constraints.
  *
  * @see ConstraintEncoderFactory
  */
-public abstract class AbstractConstraintEncoderFactory<ConstraintEncodingT> implements ConstraintEncoderFactory<ConstraintEncodingT>{
+public abstract class AbstractConstraintEncoderFactory<FormatTranslatorT extends FormatTranslator<?, ConstraintEncodingT, ?>, ConstraintEncodingT>
+            implements ConstraintEncoderFactory<ConstraintEncodingT>{
```

Comment:
should `FormatTranslatorT extends FormatTranslator<?, ConstraintEncodingT, ?>` be declared in `ConstraintEncoderFactory` instead?

Reply:
`FormatTranslatorT` is used in `AbstractConstraintEncoderFactory`, so I guess it's better to be inside `AbstractConstraintEncoderFactory`.

Reply:
keep it as is, as discussed